### ISSUE TITLE
Improve Layout and Nav with more reusable components

### ISSUE THREAD

**Issue by robertvanhoesel** [stormgateworld/web#34]
I.e. Tools and Stats use a similar header, profile kinda too but with different content.

Before coding this I'll draft up the possible layouts we want to support.

**Comment by robertvanhoesel** [comment]
WIP brain dump:
- Rely more heavily on overview pages that contain previews (recent matches + some stats, not everything on the same page)
- Add optional navigation components
- Increase reusable components so we can theoretically build pages without writing css classes


![image](https://github.com/stormgateworld/web/assets/6642554/55cf6012-d3ec-461c-b967-c69d552ecf21)

![image](https://github.com/stormgateworld/web/assets/6642554/599e7560-21a0-46f6-9218-5b6da02a5967)

(only subnav is sticky on scroll)
![image](https://github.com/stormgateworld/web/assets/6642554/05fa900f-f9fb-4201-8f9a-f29bb7eb48ee)

design can be tinted, in order to do so
![image](https://github.com/stormgateworld/web/assets/6642554/20b7c37f-8faf-475a-b90d-eb8960c37901)

```jsx
<Layout>
    {/* The Main Navigation is contained within Layout, the main nav is semi-transparent overlayed on top of the header. 
        Ideally all pages have an header */}
    <Header>{/* The Header contains the main gradient/color used for tinting */}
        <HeaderNav>{/* Optional, for including breadcrumbs as well as a list of subpages/tabs that are available */}
            <Breadcrumbs>
                
            </Breadcrumbs>
            <SubNav>

            </SubNav>
        </HeaderNav>
        <HeaderContent title={'Example Page'}>
            {/* Very simple header, always. At most some visual and text and not too much UI */}
            {/* Slot content can be used in plage of standard title */}
        </HeaderContent>
        <HeaderMeta>
            {/* Optional for 'stats' like last updated, last game data, user id, etc */}
        </HeaderMeta>
    </Header>
    <ContentSection>
        <ContentHeader/> {/* Optional, for including a title and a subtitle, as well as a description, can have slotted content and UI */}
        <Main>
            <Box/>
            <Box/>
        </Main>
        <Sidebar>
            <Box/>
            <Box/>
        </Sidebar>
    </ContentSection>
</Layout>
```

**Comment by robertvanhoesel** [comment]
More notes
- Box design drops the border and maybe even background on mobile
- Probably we want to allow box tinting too

**Comment by robertvanhoesel** [comment]
Started development in #56

**Comment by robertvanhoesel** [comment]
Closed with #61
